packages/hardhat-core/src/builtin-tasks/utils/solidity-files-cache.ts

```diff
@@ -52,30 +57,26 @@ export class SolidityFilesCache {
 
     if (result.isRight()) {
       const solidityFilesCache = new SolidityFilesCache(result.value);
-      await solidityFilesCache.removeModifiedFiles();
+      await solidityFilesCache.removeNonExistingFiles();
       return solidityFilesCache;
     }
 
-    // tslint:disable-next-line only-hardhat-error
-    throw new Error("Couldn't read cache file, try running the clean task"); // TODO use HardhatError
+    log("There was a problem reading the cache");
+
+    return new SolidityFilesCache({
```

Comment:
This will clean the cache as soon as the user upgrades, right?

Reply:
I mean, because of the change in the cache file format.

Reply:
Yes, I forgot to add a comment about this.

I think in an ideal world we would handle changes in the files format in a graceful way. For example: detecting that the file has the format v1 but the code expects format v2, and acting accordingly. But in this case, just ignoring everything and starting over is both easier and (I think) correct. We might need to change this approach in the future, but for now I think this is enough.

Reply:
agreed